Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -486,18 +486,40 @@ private func filteredAddOns(
   filteredBy shippingRule: ShippingRule?,
   baseReward: Reward
 ) -> [Reward] {
+  let isBaseRewardDigital = isRewardDigital(baseReward)
+  let isBaseRewardLocalPickup = isRewardLocalPickup(baseReward)
+
   return addOns.filter { addOn in
+    var isValidAddonToDisplay = false
+    let isAddOnDigital = isRewardDigital(addOn)
+    let isAddOnLocalPickup = isRewardLocalPickup(addOn)
+    let isAddOnLocalOrDigital = isAddOnDigital || isAddOnLocalPickup
     // For digital-only base rewards only return add-ons that are also digital-only.
-    if baseReward.shipping.enabled == false {
-      return addOn.shipping.enabled == false
+    if isBaseRewardDigital, isAddOnDigital {
+      isValidAddonToDisplay = true
+    } else if isBaseRewardLocalPickup, isAddOnLocalOrDigital {
+      isValidAddonToDisplay = true // return all addons that are digital for local base reward
+
+      if isAddOnLocalPickup {
+        if let addOnLocationId = addOn.localPickup?.id,
+          let baseRewardLocationId = baseReward.localPickup?.id,
+          addOnLocationId ==
+          baseRewardLocationId {
+          // if add on is local for local base, ensure locations are equal before displaying
+          isValidAddonToDisplay = true
+        } else {
+          isValidAddonToDisplay = false
+        }
+      }
+    } else if !isBaseRewardDigital, !isBaseRewardLocalPickup {
+      /**
+       For restricted or unrestricted shipping base rewards, unrestricted shipping
+       or digital-only add-ons are available.
```

Comment:
restricted add-ons are also available to unrestricted/restricted base rewards as long as their shipping rules intersect right?

Reply:
First we make a call to get all the location `id`'s from the shipping rules coming from `v1/shippingRules` endpoint. 

Second, the user picks/is defaulted to a shipping rule in the shipping selector. Corresponds to one of the `location` `id`'s above. Shipping selector on shown for add-ons if base reward is unrestricted/restricted.

Finally, the only add-ons available is if that add-on's `expandedShippingRules` contains a `location` `id` equal to to `location` `id` picked by the user in the selector.

Let me know if that's okay, I believe we (mobile apps) changed how the previous behavior was unrestricted/restricted base rewards.

Reply:
Also @iambchan please confirm that `shipping.preference` for any reward will not be `null`. Just because GQL Schema says it isn't a non-nullable type (ie. missing `!`)

Reply:
👍 just wanted to make sure, because the comment didn't mention returning restricted shipping rewards :) 

yup, a reward should always have a shipping preference, we don't a project to be launched without it.